### ISSUE TITLE
opam lint: warn about 1.2.0 backward-incompatible opam file

### ISSUE THREAD

**Issue by edwintorok** [ocaml/opam#2761]
I had this in my `opam` file, and when I  `opam-publish`-ed my package it failed on opam 1.2.0 with an
error about mixing bools with strings:
```
build: [ "ocaml" "pkg/pkg.ml" "build"
       "--pinned" "%{pinned}%"
       "--with-lwt" "%{lwt:installed}%"
       "--with-async" "%{async:installed}%"
       "--tests" "false"
       ]
```

The fixed version that works on both 1.2.0 and 1.2.2:
```
build: [ "ocaml" "pkg/pkg.ml" "build"
       "--pinned" pinned
       "--with-lwt" lwt:installed
       "--with-async" async:installed
       "--tests" "false"
       ]
```

Would be nice if `opam lint` or `camelus` could warn about this.

**Comment by AltGr** [comment]
@avsm, what is the state of `1.2.0` ? I'd really prefer if we coud deprecate it rather than add checks, and force current syntax to get on with old bugs...

**Comment by avsm** [comment]
@AltGr I think this is a good time to send a notice that we will deprecate OPAM 1.2.0, but I think it will sadly take quite some time to do as Debian Jessie (stable) still ships with it as the default version:

```
Get:159 http://deb.debian.org/debian/ jessie/main opam amd64 1.2.0-1+deb8u1 [1441 kB]                                                                                                        
```

So until we can get that backported or fixed, we will have to deal with OPAM 1.2.0 users. Perhaps us issuing a deprecation notice would help motivate an upstream Debian backport of 1.2.2.

**Comment by AltGr** [comment]
Maybe we could get away with making sure that `opam instal opam-devel` works on 1.2.0 :)
